aws-kendra-data-source/aws-kendra-datasource.json

```
@@ -1,105 +1,190 @@
+      "maxLength": 100,
+      "pattern": "[a-zA-Z0-9][a-zA-Z0-9_-]*"
     },
-    "ApprovalDate": {
-      "$ref": "#/definitions/InitechDateFormat"
+    "Arn": {
+      "type": "string"
     },
-    "Memo": {
-      "$ref": "#/definitions/Memo"
+    "Name": {
+      "description": "Name of data source",
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 1000,
+      "pattern": "[a-zA-Z0-9][a-zA-Z0-9_-]*"
     },
-    "SecondCopyOfMemo": {
-      "description": "In case you didn't get the first one.",
-      "$ref": "#/definitions/Memo"
+    "IndexId": {
+      "$ref": "#/definitions/IndexId"
     },
-    "TestCode": {
+    "Type": {
+      "description": "Data source type",
       "type": "string",
       "enum": [
-        "NOT_STARTED",
-        "CANCELLED"
+        "S3"
       ]
     },
-    "Authors": {
-      "type": "array",
-      "items": {
-        "type": "string"
+    "DataSourceConfiguration": {
+      "type": "object",
+      "properties": {
+        "S3Configuration": {
+          "$ref": "#/definitions/S3DataSourceConfiguration"
+        }
       }
+    },
+    "Description": {
+      "description": "Description of data source",
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 1000,
+      "pattern": "^\\P{C}*$"
+    },
+    "Schedule": {
+      "description": "Schedule",
+      "type": "string"
+    },
+    "RoleArn": {
+      "description": "Role ARN",
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 1284,
+      "pattern": "arn:[a-z0-9-\\.]{1,63}:[a-z0-9-\\.]{0,63}:[a-z0-9-\\.]{0,63}:[a-z0-9-\\.]{0,63}:[^/].{0,1023}"
     }
   },
-  "additionalProperties": false,
   "required": [
-    "TestCode",
-    "Title"
-  ],
-  "readOnlyProperties": [
-    "/properties/TPSCode"
-  ],
-  "primaryIdentifier": [
-    "/properties/TPSCode"
+    "Name",
+    "IndexId",
+    "Type",
+    "DataSourceConfiguration",
+    "RoleArn"
   ],
   "handlers": {
     "create": {
       "permissions": [
-        "initech:CreateReport"
-      ]
+        "kendra:CreateDataSource",
+        "kendra:DescribeDataSource"
+      ],
+      "timeoutInMinutes": 60
     },
     "read": {
       "permissions": [
-        "initech:DescribeReport"
-      ]
-    },
-    "update": {
-      "permissions": [
-        "initech:UpdateReport"
+        "kendra:DescribeDataSource"
       ]
     },
     "delete": {
       "permissions": [
-        "initech:DeleteReport"
-      ]
+        "kendra:DescribeDataSource",
+        "kendra:DeleteDataSource"
+      ],
+      "timeoutInMinutes": 720
     },
     "list": {
       "permissions": [
-        "initech:ListReports"
+        "kendra:ListDataSources"
       ]
     }
-  }
+  },
+  "additionalProperties": false,
+  "primaryIdentifier": [
+    "/properties/Id",
+    "/properties/IndexId"
+  ],
+  "readOnlyProperties": [
+    "/properties/Id",
+    "/properties/Arn"
+  ],
+  "createOnlyProperties": [
+    "/properties/Type",
+    "/properties/DataSourceConfiguration"
+  ]
```

Shouldn't `IndexId` be marked in [`createOnlyProperties`](https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/#resource-semantics)?
https://github.com/aws-cloudformation/cloudformation-cli/pull/604
```suggestion
  "primaryIdentifier": [
    "/properties/Id",
    "/properties/IndexId"
  ],
  "readOnlyProperties": [
    "/properties/Id",
    "/properties/Arn"
  ],
  "createOnlyProperties": [
    "/properties/IndexId",
    "/properties/Type",
    "/properties/DataSourceConfiguration"
  ]
```